### PR TITLE
Add script to reset rffe.

### PR DESCRIPTION
**Pull request by gustavosr8**

### CODE REVIEW

**Comment by gustavosr8** [comment]
Note: we need to install `expect` in the crates cpu

File: 02-cpu-configure/rffe-reset.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 <V_SLOT>"
+    exit 1
+fi
+
+V_SLOT="$1"
+RFFE_IP_ADDR="192.168.2.$(( 190 + V_SLOT ))"
+
+# Set a timeout for the telnet command (you can adjust the timeout value)
+./telnet_n_reset.expect "$RFFE_IP_ADDR"
```
Comment (henriquesimoes) [inline 02-cpu-configure/rffe-reset.sh]: This implies we need to run this in the same directory the expectation file is located. How will `rffe-reset` installed in the machine?

File: 02-cpu-configure/rffe-reset.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 <V_SLOT>"
```
Comment (henriquesimoes) [inline 02-cpu-configure/rffe-reset.sh]: Nit: usage line can be more friendly, IMHO. Being more verbose, like in `Usage: $0 <virtual-slot>` , does not seem to hurt.

File: 02-cpu-configure/rffe-reset.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
```
Comment (henriquesimoes) [inline 02-cpu-configure/rffe-reset.sh]: We don't need pipefail here, AFAIU.

File: 02-cpu-configure/rffe-reset.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+if [[ $# -ne 1 ]]; then
```
Comment (henriquesimoes) [inline 02-cpu-configure/rffe-reset.sh]: We don't need to require bash here only for this comparison. I'd make it a POSIX shell script instead by using

```shell
if [ $# -ne 1 ]; then
     ...
fi
```

and specifying `/bin/sh` in the shebang.

File: 02-cpu-configure/rffe-reset.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    echo "Usage: $0 <V_SLOT>"
+    exit 1
+fi
+
+V_SLOT="$1"
+RFFE_IP_ADDR="192.168.2.$(( 190 + V_SLOT ))"
+
+# Set a timeout for the telnet command (you can adjust the timeout value)
```
Comment (henriquesimoes) [inline 02-cpu-configure/rffe-reset.sh]: Is this a leftover comment?

**Review by henriquesimoes — CHANGES_REQUESTED** [review]
> Note: we need to install expect in the crates cpu

We can make this commit introduce such dependency by specifying it under `apt install` in ./02-cpu-configure/cpu-configure.sh.

**Review by henriquesimoes — APPROVED** [review]
Seems good. I'd wait for @ericonr's review, or even better Augusto's, before merging.

File: 02-cpu-configure/rffe-reset.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+set -eu
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <virtual_slot>"
+    exit 1
+fi
+
+expect << EOF
+spawn telnet 192.168.2.$(( 190 + $1 ))
+expect "NuttShell (NSH)"
```
Comment (ericonr) [inline 02-cpu-configure/rffe-reset.sh:11]: Shouldn't we wait for `nsh>` instead?

File: 02-cpu-configure/rffe-reset.sh [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+send "reboot\r"
```
Comment (ericonr) [inline 02-cpu-configure/rffe-reset.sh:12]: Why is the `\r` necessary?

File: 02-cpu-configure/rffe-reset.sh (hunk 6 above)
Comment (gustavosr8) [reply 02-cpu-configure/rffe-reset.sh:11]: I use [this](https://github.com/lnls-dig/bpm-cfg/blob/master/08-mmc-upload/fru_hard_rst.expect) as a reference, and in both cases it works by just looking for the first string printed in the shell after the connection is established.

File: 02-cpu-configure/rffe-reset.sh (hunk 7 above)
Comment (gustavosr8) [reply 02-cpu-configure/rffe-reset.sh:12]: According to the [expect manual](https://linux.die.net/man/1/expect):

> send [-flags] string
>     Sends string to the current process. For example, the command
> 
>     send "hello world\r"
> 
>     sends the characters, h e l l o <blank> w o r l d <return> to the current process. (Tcl includes a printf-like command (called format) which can build arbitrarily complex strings.) 
> Characters are sent immediately although programs with line-buffered input
>     will not read the characters until a return character is sent. A return character is denoted "\r".

**Review by ericonr — APPROVED** [review]

**Pull request merged**